marketplace/plugins/harperdb/lib/index.ts

```diff
@@ -0,0 +1,75 @@
+import { QueryError, QueryResult, QueryService, ConnectionTestResult } from '@tooljet-marketplace/common';
+import { SourceOptions, QueryOptions } from './types';
+import harperive from 'harperive';
+import JSON5 from 'json5';
+
+export default class Harperdb implements QueryService {
+  async run(sourceOptions: SourceOptions, queryOptions: QueryOptions, dataSourceId: string): Promise<QueryResult> {
+    const harperdbClient = await this.getConnection(sourceOptions);
+    const { operation, mode } = queryOptions;
+    let result = {};
+
+    try {
+      if (mode === 'sql') {
+        const { sql_query } = queryOptions;
+        const sql_operation_response = await harperdbClient.query(sql_query);
+        result = sql_operation_response?.data;
+      }
+
+      if (mode === 'nosql') {
+        switch (operation) {
+          case 'insert':
+            const insert_response = await harperdbClient.insert({
+              schema: queryOptions.schema,
+              table: queryOptions.table,
+              records: JSON5.parse(queryOptions.records)
+            })
+            result = insert_response.data
+            break;
+          default:
+            break;
+        }
+      }
+    } catch (err) {
+      throw new QueryError('Query could not be completed', err.error, {})
+    }
+
+    return {
+      status: 'ok',
+      data: result,
+    };
+  }
+
+  async getConnection(sourceOptions: SourceOptions): Promise<any> {
+    const { host, username, password } = sourceOptions
+
+    const DB_CONFIG = {
+      harperHost: host,
+      username: username,
+      password: password,
+      token: '',
```

Comment:
we can skip setting token and schema here since we don't provide that option at the UI end

Reply:
<img width="1026" alt="Screenshot 2023-06-16 at 5 26 12 PM" src="https://github.com/ToolJet/ToolJet/assets/40178541/c8041f9f-3771-45c1-b4d9-024de953dee7">

In Harperive SDK docs they have given either pass token / username& password, and Schema is optional. But in Typedefinition it is mandatory. So token and schema is given as Empty string.

<img width="1335" alt="Screenshot 2023-06-16 at 5 34 18 PM" src="https://github.com/ToolJet/ToolJet/assets/40178541/08bf55a0-c2c4-4b9e-acb5-6ba78871488d">

Reply:
I thought giving an empty string might make it pick those values up on authentication. Ignore my comment. 👍🏾